
<!--
The MIT License (MIT)
Copyright (c) 2016 GRVC University of Seville
Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
-->
<launch>
	<!-- Spawn a quad: send the robot XML to param server, push robot_description to factory and spawn robot in gazebo-->
    <!--include file="$(find gazebo_ros)/launch/empty_world.launch"/-->

    <!--include file="$(find px4_bringup)/launch/simulate.launch"/-->
	
    <node name="grvc_hal_quad1" pkg="grvc_quadrotor_hal" type="hal_node" respawn="false" output="screen"
		  args="-hal_ns=/mbzirc_1/hal -mavros_ns=/mavros_1 -uavId=1"/>

    <node name="grvc_hal_quad2" pkg="grvc_quadrotor_hal" type="hal_node" respawn="false" output="screen"
          args="-hal_ns=/mbzirc_2/hal -mavros_ns=/mavros_2 -uavId=2" />

    <node name="grvc_hal_quad3" pkg="grvc_quadrotor_hal" type="hal_node" respawn="false" output="screen"
          args="-hal_ns=/mbzirc_3/hal -mavros_ns=/mavros_3 -uavId=3" />

  	<node name="uav_state_machine_1" pkg="uav_state_machine" type="uav_state_machine" respawn="false" output="screen"
      args="-uavId=1"/>
    
    <node name="uav_state_machine_2" pkg="uav_state_machine" type="uav_state_machine" respawn="false" output="screen"
          args="-uavId=2"/>

    <node name="uav_state_machine_3" pkg="uav_state_machine" type="uav_state_machine" respawn="false" output="screen"
          args="-uavId=3"/>

	<!--node name="joy" pkg="joy" type="joy_node" respawn="false" output="screen">
        <param name="dev" value="/dev/input/js0"/>
        <param name="autorepeat_rate" value="2"/>
	</node-->

</launch>
